wordpress-mobile/WordPress-Android
# Captures search domain tap

**Based on:** https://github.com/wordpress-mobile/WordPress-Android/pull/19485

## Description
This PR captures the tapped domain on the domains list so that it can be used on the next steps of the flow.

Note: This PR does not link up to the Purchase screen https://github.com/wordpress-mobile/WordPress-Android/pull/19332. The intention is to capture the `ProposedDomain` information so that it can be passed in the https://github.com/wordpress-mobile/WordPress-Android/issues/19382 implementation for testing adding the domain to the cart and opening the checkout webview.

## To test
1. Make sure you have enabled the domain_management feature flag.
2. Go to Me -> Domains
3. Tap on "+" button on the toolbar
4. Search for a domain
5. Tap on one of the domain results
6. **Verify** that the `🔵 Tracked: domain_management_search_domain_tapped, Properties: {"domain_name":"tapped.domain"}` tracking event is emitted 

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
Added new test cases in `NewDomainSearchViewModelTest`

PR submission checklist:


UI Changes testing checklist:
